src/mad/sh/scripts_common.sh

```diff
@@ -577,25 +609,63 @@ set -e -o pipefail
 if [ ! -d $2 ]; then
     mkdir -p $2
 fi
+EOF`
+    SSH_EXEC_RC=$?
 
-# create or update .monitor content
-if [ -n "$3" ]; then
-    MONITOR_FN="\$(dirname $2)/.monitor"
+    if [ $SSH_EXEC_RC -ne 0 ]; then
+        error_message "Error creating directory $2 at $1: $SSH_EXEC_ERR"
 
-    MONITOR=''
-    if [ -f "\\${MONITOR_FN}" ]; then
-        MONITOR="\\$(cat "\\${MONITOR_FN}" 2>/dev/null)"
+        exit $SSH_EXEC_RC
     fi
+}
 
-    if [ "x\\${MONITOR}" != "x$3" ]; then
-        echo "$3" > "\\${MONITOR_FN}"
-    fi
+# Enables local monitoring for host $2 and DS $1, by creating the corresponding
+# .monitor file containing the $3 ( the driver name) (e.g ssh)
+function enable_local_monitoring
+{
+    SSH_EXEC_ERR=`$SSH $1 bash -s 2>&1 1>/dev/null <<EOF
+set -e -o pipefail
+
+MONITOR_FN="\$(dirname $2)/.monitor"
+
+# create or update .monitor content
+MONITOR=''
+if [ -f "\\${MONITOR_FN}" ]; then
+    MONITOR="\\$(cat "\\${MONITOR_FN}" 2>/dev/null)"
+fi
+
+if [ "x\\${MONITOR}" != "x$3" ]; then
+    echo "$3" > "\\${MONITOR_FN}"
 fi
 EOF`
+
     SSH_EXEC_RC=$?
 
     if [ $SSH_EXEC_RC -ne 0 ]; then
-        error_message "Error creating directory $2 at $1: $SSH_EXEC_ERR"
+        error_message "Error creating $2/.monitor at $1: $SSH_EXEC_ERR"
+
+        exit $SSH_EXEC_RC
+    fi
+}
+
+# Disables local monitoring for host $2 and DS $1, by removing the corresponding
+# .monitor file
+function disable_local_monitoring
+{
+    SSH_EXEC_ERR=`$SSH $1 bash -s 2>&1 1>/dev/null <<EOF
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The use of backticks for command substitution in shell scripts can lead to vulnerabilities, quoting issues and nested command complexities. To mitigate this issue, it is recommended to use the more secure &#39;$(...)&#39; syntax for command substitution. This method properly quotes the substituted command, which is more readable, easier to nest, and avoids undefined behaviors related to quoting. Adopting this practice improves code clarity and maintainability. For more information on shell scripting best practices, refer to the GNU Bash Manual (https://www.gnu.org/software/bash/manual/bash.html#Command-Substitution).